File: app/src/main/java/edu/gatech/cs2340/Marketplace.java
package edu.gatech.cs2340;

import java.io.Serializable;
import java.util.Random;

class Marketplace implements Serializable {
    private int MTLP;
    private int BasePrice;
    private int IPL;
    private int Var;
    private int level;
    private static final int waterP = 30;
    private static final int fursP = 250;
    private static final int foodP = 100;
    private static final int oreP = 350;
    private static final int gamesP = 250;
    private static final int firearmsP = 1250;
    private static final int machinesP = 900;
    private static final int medicinesP = 650;
    private static final int narcoticsP = 1500;
    private static final int robotsP = 3000;

    private static final int waterIPL = 3;
    private static final int fursIPL = 25;
    private static final int foodIPL = 10;
    private static final int oreIPL = 35;
    private static final int gamesIPL = -25;
    private static final int firearmsIPL = -125;
    private static final int machinesIPL = -90;
    private static final int medicinesIPL = -65;
    private static final int narcoticsIPL = -150;
    private static final int robotsIPL = -300;
    private static final int var = 150;

    private int getMTLP() { return this.MTLP; }

    private int getBasePrice() { return this.BasePrice;}

    private int getIPL() { return this.IPL;}

    private int getVar() { return this.Var;}

    void techLevel(String string) {
        level = -1;
        if ("Pre-Agriculture".equals(string)) {
            level = 0;
        }
        if ("Agriculture".equals(string)) {
            level = 1;
        }
        if ("Medieval".equals(string)) {
            level = 2;
        }
        if ("Renaissance".equals(string)) {
            level = 3;
        }
        if ("Early Industrial".equals(string)) {
            level = 4;
        }
        if ("Industrial".equals(string)) {
            level = 5;
        }
        if ("Post-Industrial".equals(string)) {
            level = 6;
        }
        if ("Hi-Tech".equals(string)) {
            level = 7;
        }

     }

    int getLevel() {
        return level;
    }

    void Water() {
        MTLP = 0;
        BasePrice = waterP;
        IPL = waterIPL;
        Var = 4;
    }

    void Furs() {
        MTLP = 0;
        BasePrice = fursP;
        IPL = fursIPL;
        Var = 10;
    }

    void Food() {
        MTLP = 1;
        BasePrice = foodP;
        IPL = foodIPL;
        Var = 5;
    }

    void Ore() {
        MTLP = 2;
        BasePrice = oreP;
        IPL = oreIPL;
        Var = 10;
    }

    void Games() {
        MTLP = 3;
        BasePrice = gamesP;
        IPL = gamesIPL;
        Var = 5;
    }

    void Firearms() {
        MTLP = 3;
        BasePrice = firearmsP;
        IPL = firearmsIPL;
        Var = 100;
    }

    void Medicine() {
        MTLP = 4;
        BasePrice = medicinesP;
        IPL = medicinesIPL;
        Var = 10;
    }

    void Machines() {
        MTLP = 4;
        BasePrice = machinesP;
        IPL = machinesIPL;
        Var = 5;
    }

    void Narcotics() {
        MTLP = 5;
        BasePrice = narcoticsP;
        IPL = narcoticsIPL;
        Var = var;
    }

    void Robots() {
        MTLP = 6;
        BasePrice = robotsP;
        IPL = robotsIPL;
        Var = 100;
    }

    int calculatePrice() {
        Random rand = new Random();
        int buffer = (rand.nextInt(getVar()) + 1);
        return getBasePrice() + (getIPL() * (level - getMTLP())) + (buffer);
    }
}

File: app/src/main/java/edu/gatech/cs2340/Marketplace.kt
package edu.gatech.cs2340

import java.io.Serializable
import java.util.Random

internal class Marketplace : Serializable {
    private var mtlp: Int = 0
    private var basePrice: Int = 0
    private var ipl: Int = 0
    private var Var: Int = 0
    var level: Int = 0
        private set

    private fun getVar(): Int {
        return this.Var
    }

    fun techLevel(string: String?) {
        level = -1
        if ("Pre-Agriculture" == string) {
            level = 0
        }
        if ("Agriculture" == string) {
            level = 1
        }
        if ("Medieval" == string) {
            level = 2
        }
        if ("Renaissance" == string) {
            level = 3
        }
        if ("Early Industrial" == string) {
            level = 4
        }
        if ("Industrial" == string) {
            level = 5
        }
        if ("Post-Industrial" == string) {
            level = 6
        }
        if ("Hi-Tech" == string) {
            level = 7
        }

    }

    fun Water() {
        mtlp = 0
        basePrice = waterP
        ipl = waterIPL
        Var = 4
    }

    fun Furs() {
        mtlp = 0
        basePrice = fursP
        ipl = fursIPL
        Var = 10
    }

    fun Food() {
        mtlp = 1
        basePrice = foodP
        ipl = foodIPL
        Var = 5
    }

    fun Ore() {
        mtlp = 2
        basePrice = oreP
        ipl = oreIPL
        Var = 10
    }

    fun Games() {
        mtlp = 3
        basePrice = gamesP
        ipl = gamesIPL
        Var = 5
    }

    fun Firearms() {
        mtlp = 3
        basePrice = firearmsP
        ipl = firearmsIPL
        Var = 100
    }

    fun Medicine() {
        mtlp = 4
        basePrice = medicinesP
        ipl = medicinesIPL
        Var = 10
    }

    fun Machines() {
        mtlp = 4
        basePrice = machinesP
        ipl = machinesIPL
        Var = 5
    }

    fun Narcotics() {
        mtlp = 5
        basePrice = narcoticsP
        ipl = narcoticsIPL
        Var = `var`
    }

    fun Robots() {
        mtlp = 6
        basePrice = robotsP
        ipl = robotsIPL
        Var = 100
    }

    fun calculatePrice(): Int {
        val rand = Random()
        val buffer = rand.nextInt(getVar()) + 1
        return basePrice + ipl * (level - mtlp) + buffer
    }

    companion object {
        private val waterP = 30
        private val fursP = 250
        private val foodP = 100
        private val oreP = 350
        private val gamesP = 250
        private val firearmsP = 1250
        private val machinesP = 900
        private val medicinesP = 650
        private val narcoticsP = 1500
        private val robotsP = 3000

        private val waterIPL = 3
        private val fursIPL = 25
        private val foodIPL = 10
        private val oreIPL = 35
        private val gamesIPL = -25
        private val firearmsIPL = -125
        private val machinesIPL = -90
        private val medicinesIPL = -65
        private val narcoticsIPL = -150
        private val robotsIPL = -300
        private val `var` = 150
    }
}

